src/Resources/app/storefront/src/checkout/confirm-order.plugin.js

```diff
@@ -142,15 +148,120 @@ export default class ConfirmOrderPlugin extends Plugin {
     handlePaymentAction(paymentAction) {
         try {
             const paymentActionResponse = JSON.parse(paymentAction);
-            if (!!paymentActionResponse.action) {
-                window.adyenCheckout.createFromAction(paymentActionResponse.action).mount('[data-adyen-payment-action-container]');
-                return;
-            }
             if (paymentActionResponse.isFinal) {
                 location.href = window.returnUrl;
             }
+            if (!!paymentActionResponse.action) {
+                window.adyenCheckout
+                    .createFromAction(paymentActionResponse.action)
+                    .mount('[data-adyen-payment-action-container]');
+            }
         } catch (e) {
             console.log(e);
         }
     }
+
+    initializeCustomPayButton() {
+        // get selected payment method
+        let selectedAdyenPaymentMethod = this.getSelectedPaymentMethodKey();
+        if (!adyenConfiguration.componentsWithPayButton.includes(selectedAdyenPaymentMethod)) {
+            return;
+        }
+
+        const ADYEN_CHECKOUT_CONFIG = Object.assign(
+            {},
+            adyenCheckoutConfiguration,
+            {paymentMethodsResponse: JSON.parse(adyenCheckoutConfiguration.paymentMethodsResponse)}
+        );
+        let adyenCheckout = new AdyenCheckout(ADYEN_CHECKOUT_CONFIG);
+        // get selected payment method object
+        let selectedPaymentMethod = adyenCheckout.paymentMethodsResponse.paymentMethods
+            .filter(item => item.type == selectedAdyenPaymentMethod);
+
+        if (selectedPaymentMethod.length < 1) {
+            return;
+        }
+        let selectedPaymentMethodObject = selectedPaymentMethod[0];
+
+        const createPayButton = function (response) {
+            let cart = JSON.parse(response);
+
+            const PAY_BUTTON_CONFIG = Object.assign({}, selectedPaymentMethodObject, {
+                showPayButton: true,
+                amount: {
+                    value: cart.price.totalPrice * 100,
+                    currency: adyenCheckoutOptions.currency,
+                },
+                onClick: (resolve, reject) => {
+                    let tos = $('input#tos');
```

Comment:
isn't there a validate function provided by shopware that is used to validate the form when clicking on the original pay button?

Reply:
If there is one, I didn't find it :(